DEVDOC.md

```diff
@@ -313,3 +313,60 @@ A few useful homemade packages:
 ### Java
 
 > TODO
+
+## Code generated at compile time
+
+A big amount of code is generated at compile time. Code is typically generated
+via APIC (API Compiler, `cmd/apic/`). APIC takes as input `.api` and `.tmpl`
+files, and it generates code in another language like Go, C++ or Java.
```

Comment:
```suggestion
A large amount of code is generated at compile time. Code is typically generated
via APIC (API Compiler, `cmd/apic/`). APIC takes as input `.api` and `.tmpl`
files, and generates code using go's [templating system](https://golang.org/pkg/text/template/).
```

Technically, APIC can generate other things than code and we don't actually generate any Java code. I don't think listing the languages helps here.

Reply:
OK for "large", not for the templating language: it is go templating systems with several AGI-sepcific extensions, so I leave it to the description of `.tmpl` files which comes further down.